### PR TITLE
Check for Inert and Vision as dependencies

### PR DESCRIPTION
**Pull request by osukaa**
I'm still debating about how to handle hapi < 9

### CODE REVIEW

**Comment by devinivy** [comment]
Could hapi < 9 users just use the old version of tv?  Maybe for the time being 4.x and (presumed to-be) 5.x series of tv could both be maintained with the same patch fixes, until hapi 8 is no longer supported by the hapi team.

**Comment by osukaa** [comment]
I think that would be an appropriate solution!

**Comment by osukaa** [comment]
In v8 they're registered in the core and beginning v9 it would ask the user to add them since it's in the `server.dependency`

**Comment by arb** [comment]
Might be worth taking a pass through "package.json" and making sure the dependency versions are cleaned up. I believe the standard is `#.x.x` for hapijs packages and `#.#.x` for non-hapijs package.

File: package.json [hunk 1]
```diff
@@ -20,15 +20,17 @@
     "handlebars": "3.X.X",
     "hbsfy": "2.X.X",
     "hoek": "2.X.X",
+    "inert": "^3.0.1",
```
Comment (arb) [inline package.json]: Aren't these technically devDependencies? inert and vision?

File: test/index.js [hunk 2]
```diff
@@ -371,20 +393,51 @@ it('uses specified route prefix for assets', function (done) {
...
+        handler: function (request, reply) {
+
+            return reply('1');
+        }
+    });
+
+    server.register({ register: Tv, options: { port: 0 } }, function (err) {
+
+        expect(err).to.not.exist();
+        try {
```
Comment (arb) [inline test/index.js]: I think you can do `expect(server.start(function {}).to.throw()`

https://github.com/hapijs/code#throwtype-message

Comment (osukaa) [inline test/index.js]: I tried that but it doesn't actually catch the error so I get a failed test.

File: package.json (hunk 1 above)
Comment (osukaa) [inline package.json]: You're right, I'm moving them to the devDependency section.

**Comment by osukaa** [comment]
@arb I cleaned the `package.json`

**Comment by arb** [comment]
Looks good. You should probably have one more person check it thought just to be safe.

**Comment by osukaa** [comment]
Maybe @geek can help us out. :) 



—

File: test/index.js (hunk 2 above)
Comment (ldesplat) [inline test/index.js]: Do it like this:
```
var throws = function () {

    server.start(function () {});
};

expect(throws).to.throw();
```

The last last line could also be `expect(throws).to.throw(Error, 'Plugin tv missing dependency inert in connection');`

Comment (osukaa) [inline test/index.js]: Thanks man! That fixed the issue :D

**Comment by Marsup** [comment]
Hello, just curious, why did you set the minor versions on most of your dependencies ?

**Comment by osukaa** [comment]
@Marsup I thought about the standard that @arb mentioned up there.

**Comment by Marsup** [comment]
Oh thanks, I didn't see it. @arb where's that mentioned ?

**Comment by arb** [comment]
Do conversations with Eran count :-) You're right though, I don't THINK that it's documented anywhere. I think the rules are:

- #.x.x for hapijs packages
- #.#.x for non-hapijs packages with version >=1.0.0
- #.#.# for non-hapijs packages with version < 1.0.0 (if you use them at all)

**Comment by lloydbenson** [comment]
Maybe a good opportunity to add it in the contrib repo?

**Pull request closed without merging**